File: ueims_frontend/src/services/DashboardService.ts
import { api } from './api';

export const DashboardService = {
  getCommandCenterSummary: async () => {
    const response = await api.get('/dashboard/command-center-summary');
    return response.data.result;
  },

  getEmploymentRateChart: async (semesterId: string) => {
    const response = await api.get(`/dashboard/employment-rate/${semesterId}`);
    return response.data;
  },
  getMajorDistributionChart: async (semesterId: string) => {
    const response = await api.get(`/dashboard/major-distribution/${semesterId}`);
    return response.data;
  },
  getGradeDistributionChart: async (semesterId: string) => {
    const response = await api.get(`/dashboard/grade-distribution/${semesterId}`);
    return response.data;
  },
  getInterviewPassRateChart: async (semesterId: string) => {
    const response = await api.get(`/dashboard/interview-pass-rate/${semesterId}`);
    return response.data;
  },
  getAverageRatingChart: async (semesterId: string) => {
    const response = await api.get(`/dashboard/average-rating/${semesterId}`);
    return response.data;
  },
};

File: ueims_backend/src/main/java/com/ueims/service/DashboardService.java
package com.ueims.service;

import java.util.List;
import java.util.UUID;

import com.ueims.model.dto.dashboard.ChartDataDTO;
import com.ueims.model.dto.dashboard.CommandCenterSummaryDTO;

public interface DashboardService {
    CommandCenterSummaryDTO getCommandCenterSummary();

    List<ChartDataDTO> getEmploymentRateChart(UUID semesterId);

    List<ChartDataDTO> getInterviewPassRateChart(UUID semesterId);

    List<ChartDataDTO> getMajorDistributionChart(UUID semesterId);

    List<ChartDataDTO> getGradeDistributionChart(UUID semesterId);

    List<ChartDataDTO> getAverageRatingChart(UUID semesterId);
}
